scheduler/src/helper/generator.helper.ts

```diff
@@ -0,0 +1,423 @@
+    const client = getClient();
+    const responsibility = getResponsibility();
+    const serviceLine = getServiceLine();
+    const stob = getStob();
+    const project = getProject();
+    const location = getLocation();
+    const future = CgiConstants.FUTURE;
+    const unusedFiller = ' '.repeat(16);//16 spaces
+    let result = '';
+    result = result + client + responsibility + serviceLine + stob + project + location + future + unusedFiller;
+    return result;
+  }
+  
+  function getClient(): string {
+    // 3 chars
+    return CgiConstants.CLIENT;
+  }
+  
+  function getResponsibility(): string {
+    // 5 chars
+    return CgiConstants.RESPONSIBILITY;
+  }
+  
+  function getServiceLine(): string {
+    // 5 chars
+    return CgiConstants.SERVICE_LINE;
+  }
+  
+  function getStob(): string {
+    // 4 chars
+    return CgiConstants.STOB;
+  }
+  
+  function getProject(): string {
+    // 7 chars
+    return CgiConstants.PROJECT;
+  }
+  
+  function getLocation(): string {
+    // 6 chars
+    return `0`.repeat(6);
+  }
+  
+  function getFuture(): string {
+    // 4 chars
+    return `0`.repeat(4);
+  }
+  
+  function getUnusedFiller(): string {
+    // 16 chars
+    return `0`.repeat(16);
+  }
+  
+  function getLineCode(): string {
+    // 1 char, C or D
+    return CgiConstants.LINE_CODE;
+  }
+  
+  function getLineDescription(): string {
+    // 100 chars
+    return ` `.repeat(100);
+  }
+  
+  function getFeederNumberClientSystem(): string {
+    // 4 chars
+    return `0`.repeat(4);
+  }
+  
+  function getControlCount(transactions: Transaction[]): string {
+    // 15 chars
+    const numberStr = transactions.length.toString();
+    const desiredLength = 15;
+    const paddedString = numberStr.padStart(desiredLength, '0');
+    return paddedString;
+    
+  }
+  
+  function populateBatchHeader(filename: string, ackFilename: string): string {
+    let batchHeader: string = ``;
+    const feederNumber: string = process.env.FEEDER_NUMBER;
+    const batchType: string = CgiConstants.BATCH_TYPE;
+    const transactionType: string = CgiConstants.TRANSACTION_TYPE_BH;
+    const delimiterHex = 0x1D;
+    const delimiter = String.fromCharCode(delimiterHex);
+    const fiscalYear: number = getFiscalYear();
+    const batchNumber: string = getBatchNumber();
+    const messageVersion: string = CgiConstants.MESSAGE_VERSION;
+    batchHeader = batchHeader + feederNumber + batchType + transactionType + delimiter + feederNumber + fiscalYear + batchNumber + messageVersion + delimiter + `\n`;
+    const bt: BatchHeader = new BatchHeader();
+    bt.feederNumber = feederNumber;
+    bt.batchType = batchType;
+    bt.transactionType = transactionType;
+    bt.fiscalYear = fiscalYear;
+    bt.batchNumber = batchNumber;
+    bt.messageVersion = messageVersion;
+    console.log(filename);
+    console.log(ackFilename);
+    return batchHeader;
+  }
+
+function populateJournalHeader(transactions: Transaction[]): string {
```

Comment:
Can you write documentation on all of these functions and provide documentation on the general flow of execution? Looking at this PR, I've no idea what any of these steps even do.